spark/interpreter/src/main/java/org/apache/zeppelin/spark/SparkVersion.java

```diff
@@ -86,7 +86,11 @@ public boolean isSpark2() {
   }
 
   public boolean isSecretSocketSupported() {
-    return this.newerThanEquals(SPARK_2_3_1);
+    return this.newerThanEquals(SPARK_2_3_1) ||
+            (this.newerThanEquals(SparkVersion.fromVersionString("2.2.2")) &&
+                    this.olderThan(SparkVersion.fromVersionString("2.3.0"))) ||
+            (this.newerThanEquals(SparkVersion.fromVersionString("2.1.3")) &&
+                    this.olderThan(SparkVersion.fromVersionString("2.2.0")));
```

Comment:
I think this works, but if and only if they are evaluated in the given order, which is a bit error prone. might be worthwhile to add a `newerThanEqualsPatchVersion` for `2.2.[2+]` type of check